dask_geomodeling/raster/sources.py

```diff
@@ -14,7 +14,117 @@
 __all__ = ["MemorySource", "RasterFileSource"]
 
 
-class MemorySource(RasterBlock):
+class RasterSourceBase(RasterBlock):
+
+    @staticmethod
+    def process(process_kwargs):
+        mode = process_kwargs["mode"]
+
+        # handle empty requests
+        if mode == "empty_vals":
+            return
+        elif mode == "empty_time":
+            return {"time": []}
+        elif mode == "empty_meta":
+            return {"meta": []}
+
+        bands = process_kwargs["bands"]
+        length = bands[1] - bands[0]
+
+        # handle time requests
+        if mode == "time":
+            start = process_kwargs["start"]
+            delta = process_kwargs["delta"]
+            return {"time": [start + i * delta for i in range(length)]}
+
+        # open the dataset
+        source = process_kwargs.get("dataset")
```

Comment:
This is where the two paths (memorysource and filesource) come together. Can you annotate that?